### ISSUE TITLE
Error in validateInput useEffect Error: validateInput expected an inputEl.form to be available for input

### ISSUE THREAD

**Issue by CSFlorin** [brophdawg11/remix-validity-state#39]
Trying to access the value of one input inside another input component using `useValidatedInput` but when you call it without destructuring `getInputAttrs` inside an input element, you get that error. I'd just like to read the `info`. Should this work?

**Comment by CSFlorin** [comment]
Well turns out you don't get the value that way anyway. Looks like I've just gotta use refs anyway.

**Comment by brophdawg11** [comment]
👋 Thanks for trying out `remix-validity-state` @CSFlorin!

> when you call it without destructuring getInputAttrs inside an input element, you get that error

I'm not quite sure the use case you're describing that produced that error - would you be able to provide a reproduction?  Seeing a use-case in action would help understand the true source of the error and what you're trying to accomplish.

> It would be nice if info contained the value.  It would be nicer if there was also a function returned to update a field

I think it's probably safe to expose `value` on `info`.  And that would even open up a potential API change to a `serverFormInfo.submittedFormValues` entirely since we'd be able to send back the submitted values inside `serverFormInfo.inputs`, but I need to think about that one a bit more and see if it has any rough edges.  

I'm a bit less confident that we want to expose a `setValue` since I'm generally aiming to move away from controlled inputs in this library and leverage built-in HTML form functionality as much as possible.  If we expose the `value` and the `ref` (or if you used your own ref), then you'd be able to programmatically set another inputs value if needed via the following:

```js
  // Assume we expose `value` and `ref` from here
  let a = useValidatedInput({ name: "a"  });
  let b = useValidatedInput({ name: "b" });
  
  React.useEffect(() => {
    if (!a.ref?.current || !b.ref?.current) return;
    b.ref.current.value = calculateValueBasedOnA(a.ref.current.value);
    b.ref.current.dispatchEvent(new Event("input"));
  }, [a, b]);
```

> like a hidden input that stays in sync with the visible one

I'm also curious about this hidden input use-case - presumably if it's hidden then it's something you just need at submission time, so ifs there a reason you couldn't just just generate the derived value in your action based on the submitted value of a field?  Instead of of constantly deriving the value and sticking it in a hidden input client side?

Also if it's a hidden input I thin you could "sync" it with another field given the `value` above by just using `defaultValue`?

```jsx
<input {...b.getInputAttrs()} defaultValue={calculateValueBasedOnA(a.value)} />
```

**Comment by CSFlorin** [comment]
Right now there isn't a clear way to even pass in my own ref though so I think exposing it would be an essential minimum. If I pass in my own, it overwrites the one coming from `getInputAttributes()` and I get the same error from the library.

Stackblitz isn't working for me right now to show an example, but if you call `useValidatedInput` without passing the result of `getInputAttributes()` to an input, you get this error.

The value I'm deriving the hidden input value from is only available in the component via the loader data so it isn't available in the action.

**Comment by brophdawg11** [comment]
You _should_ be able to pass a ref via `useValidatedInput` - that got implemented a bit ago in https://github.com/brophdawg11/remix-validity-state/pull/19.  If that's not working that's news to me - I think that code got copy pasted from another library so I sort of assumed it worked as advertised :).  I'll take a look into that.  

> I think the defaultValue thing should work, but this brings us back to being able to access the input values, which this library seems to actively prevent

I was suggesting above that we expose the value for this use case.  It's the exposure of setValue and elimination of submittedValues that I'm not yet sure about.

**Comment by CSFlorin** [comment]
Oh I didn't realize you pass it into the hook 🤦‍♂️ Alrighty I'll give it all a spin. Thanks for the responsiveness, and yes the value would be cleaner but this should unblock.

**Comment by brophdawg11** [comment]
Released in `0.10.0`
